hwilib/serializations.py

```diff
@@ -758,6 +758,7 @@ def __init__(self, tx: Optional[CTransaction] = None) -> None:
         self.inputs: List[PartiallySignedInput] = []
         self.outputs: List[PartiallySignedOutput] = []
         self.unknown: Dict[bytes, bytes] = {}
+        self.global_xpubs: Dict[bytes, bytes] = {}
```

Comment:
I think this is supposed to be `Dict[bytes, list[int]]`.

I think it would be better to actually flip this around though. The proposed implementation to Bitcoin Core has the global xpubs as a multimap. So the key is the derivation path, and the value is a list of xpubs for that derivation path (because there could be collisions). The type ends up being `Dict[list[int], list[bytes]]`. I think doing it this way makes it easier later when we want to do lookups by derivation path.

Reply:
Ok, but as it's not possible to use `list` as a key for `dict`, I moved unpacking of the derivation outside and kept it as bytes in the PSBT deser, so the type is `Dict[bytes, list[bytes]]`.